integration/valkey_search_test_case.py

```diff
@@ -54,3 +57,157 @@ def parse_valkey_info(self, section):
+        for key, value in d.items():
+            result.append(f"--{key}")
+            result.append(str(value))
+        return " ".join(result)
+
+    def _start_server(self, port):
+        server_path = os.getenv("VALKEY_SERVER_PATH")
+        pid = os.getpid()
+        testdir = f"/tmp/valkey-search-clusters/{pid}/shard-{port}"
+        os.makedirs(testdir, exist_ok=True)
+        curdir = os.getcwd()
+        os.chdir(testdir)
+        lines = [
+            "enable-debug-command yes",
+            f"loadmodule {os.getenv('JSON_MODULE_PATH')}",
+            f"loadmodule {os.getenv('MODULE_PATH')} --use-coordinator",
+            f"dir {testdir}",
+            "cluster-enabled yes",
+        ]
+        with open(f"{testdir}/valkey.conf", "w+") as f:
+            for line in lines:
+                f.write(f"{line}\n")
+            f.write("\n")
+            f.close()
+
+        server, client = self.create_server(
+            testdir=testdir,
+            server_path=server_path,
+            args="",
+            port=port,
+            conf_file=f"{testdir}/valkey.conf",
+        )
+        os.chdir(curdir)
+        logfile = f"{testdir}/logfile_{port}"
+        return server, client, logfile
+
+    def _split_range_pairs(self, start, end, n):
+        points = [start + i * (end - start) // n for i in range(n + 1)]
+        return list(zip(points[:-1], points[1:]))
+
+    def add_slots(self, node_idx, first_slot, last_slot):
+        client: Valkey = self.client_for_primary(node_idx)
+        slots_to_add = list(range(int(first_slot), int(last_slot)))
+        try:
+            client.execute_command("CLUSTERADMIN ADDSLOTS", *slots_to_add)
+        except Exception as e:
+            client.execute_command("CLUSTER ADDSLOTS", *slots_to_add)
+
+    def cluster_meet(self, node_idx, primaries_count):
+        client: Valkey = self.client_for_primary(node_idx)
+
+        for node_to_meet in range(0, primaries_count):
+            if node_to_meet == node_idx:
+                continue
+            meet_command = (
+                f"MEET 127.0.0.1 {self.get_primary_port(node_to_meet)}"
+            )
+            try:
+                client.execute_command(
+                    " ".join(
+                        [
+                            "CLUSTERADMIN",
+                            "MEET",
+                            "127.0.0.1",
+                            f"{self.get_primary_port(node_to_meet)}",
+                        ]
+                    )
+                )
+            except Exception as e:
+                client.execute_command(
+                    " ".join(
+                        [
+                            "CLUSTER",
+                            "MEET",
+                            "127.0.0.1",
+                            f"{self.get_primary_port(node_to_meet)}",
+                        ]
+                    )
+                )
+
+    @pytest.fixture(autouse=True)
+    def setup_test(self, setup):
```

Comment:
Instead of using a fixture with predefined configurations (in this case, singe shard, one primary, no replicas), it would be good if this interface allows configurability in terms number of shards and number of replicas per shard

Reply:
Ack. Thank you. Some of the issues seen so far are during fanned out requests to remote nodes - So this testing will be useful :D